base_report_docx/report/report_docx.py

```diff
@@ -55,20 +55,29 @@ def create_source_docx(self, cr, uid, ids, context=None):
             'pdf': 'report.pdf',
             'docx': 'report.docx'
         }
+        file = tmp_folder_name + output_report[output_type]
         self._delete_temp_folder(tmp_folder_name)
         self._create_temp_folder(tmp_folder_name)
 
         self._generate_reports(
             cr, uid, context, tmp_folder_name, data,
             output_type, output_report)
 
-        report = self._get_convert_file(
-            tmp_folder_name, output_report[output_type])
-
+        report = self._get_convert_file(file)
+        # this is a hook to perform additional operations on the file
+        # before it's deleted
+        self.on_delete_output_file(cr, uid, ids, file, context)
         self._delete_temp_folder(tmp_folder_name)
 
         return (report, output_type)
 
+    def on_delete_output_file(cr, uid, ids, file, context):
+        """
+            Override this method to save the pdf on your own server
```

Comment:
Any operation can be performed in this hook. This comment is way too specific. Replace with:
"Override this method to perform operations on the output file before it gets deleted. For instance, you can copy it to another location in the file system."